### ISSUE TITLE
feature: implement `toContainElement` Jest matcher

### ISSUE THREAD

**Issue by mdjastrzebski** [callstack/react-native-testing-library#1486]
## Describe the Feature

[`toContainElement`](https://github.com/testing-library/jest-dom#tocontainelement) is a Jest DOM matcher asserting that given element contains other passed element.

We also have [`toContainElement`](https://github.com/testing-library/jest-native/blob/main/src/to-contain-element.ts) already implemented in Jest Native. So migrating it to current repo, updating code to use RNTL helpers and reviewing TS types + tests quality should be the scope for the task.

## Possible Implementations
Proposed API:

```ts
export function toContainElement(
  this: jest.MatcherContext,
  container: ReactTestInstance,
  element: ReactTestInstance | null,
) {
```

The matcher should:
1) Validate that it is invoked on a host element
2) Check if given element contains the passed element

Each matcher should have a fairly comprehensive test suite.

## Related Issues
#1454

**Comment by mdjastrzebski** [comment]
Resolved by #1495

**Comment by mdjastrzebski** [comment]
🎉 Release in version: [v12.4.0](https://github.com/callstack/react-native-testing-library/releases/tag/v12.4.0)
